### ISSUE TITLE
AI skips turns and loses all its army

### ISSUE THREAD

**Issue by Nordsoft91** [vcmi/vcmi#2647]
**Describe the bug**
Starting from 123 Ai skips several turns, after that it captured some troops but started to fight very risk and lost almost all army prior 135. Maybe not a bug, but still such behavior is not good

NeutralAI: StupidAI
EnemyAI: Battle AI + NKAI

Saves attached: Test1 - 123, Test2 - 135
[NKAI_behavior.zip](https://github.com/vcmi/vcmi/files/12413611/NKAI_behavior.zip)

**Comment by Nordsoft91** [comment]
UPD: on 411 i captured enemy town, and it obviously doesn't buy troops at all. I didn't do anything and collected 60k gold (town +500 gold) while enemy built everything, including capitol and I'm sure it has enough money.

Screenshot shown troops to hire

<img width="929" alt="Снимок экрана 2023-08-23 в 03 25 36" src="https://github.com/vcmi/vcmi/assets/9308612/9ff18142-8f81-4aa9-8618-f304f5d17cf2">

**Comment by Nordsoft91** [comment]
@nullkiller you might be interested

**Comment by nullkiller** [comment]
Looks like AI can retreat from neutrals. I tried 135 save and the hero was recruited again afterwards.
After restart it captured second town in 7 days. On 243 or something it came to me.
![image](https://github.com/vcmi/vcmi/assets/5644734/b5f84a16-41a0-47d7-85be-d5a80a0ca93f)

**Comment by nullkiller** [comment]
I agree it plays a bit too risky but this is how it is designed. Map forces to play risky. Too strong guards.

**Comment by nullkiller** [comment]
I loaded 123 save and I see it attacked monster and likely retreated. Maybe I need to adjust safe attack constant. I will take more look onto it later

**Comment by C3pa** [comment]
Is this any better with Nullkiller2?

**Comment by dydzio0614** [comment]
should be better with VCMI 1.6 already

**Comment by GrafDeVan** [comment]
The AI ​​skipping turns for various reasons is still a thing, but not buying an army shouldn't be.
@Nordsoft91, have you tried the latest releases?

**Comment by GrafDeVan** [comment]
@GeorgeK1ng Looks like this is no longer relevant. A fresh save is needed.
